### PR TITLE
Qa dashboard service liveness code quality

### PR DESCRIPTION
**Pull request by tisnik**
Pls. see the first version 'live': https://fabric8-analytics.github.io/dashboard/dashboard.html

### CODE REVIEW

File: dashboard/src/dashboard.py [hunk 1]
```diff
@@ -97,22 +105,89 @@ def run_docstyle_check(repository):
...
+
+    files = {}
+    passed = 0
+    failed = 0
+
+    with open(filename) as fin:
+        for line in fin:
+            if line.endswith(".py\n"):
+                source = line.strip()
+            if line.endswith("    Pass\n"):
```
Comment (jpopelka) [inline dashboard/src/dashboard.py]: What about adding `line = line.strip()` as a first line of this `for` loop and then you don't have to check whether the line endswith `\n`.

File: dashboard/src/dashboard.py [hunk 2]
```diff
@@ -97,22 +105,89 @@ def run_docstyle_check(repository):
...
+              "> ../{repo}.count;popd".format(repo=repository)
+    os.system(command)
+    files = {}
+    count = 0
+
+    with open("{repo}.count".format(repo=repository)) as fin:
+        for line in fin:
+            count += 1
+
+    results.source_files[repository] = {"count": count}
```
Comment (jpopelka) [inline dashboard/src/dashboard.py]: Perhaps also

```python
os.remove("{repo}.count".format(repo=repository))`
```

File: dashboard/src/dashboard.py [hunk 3]
```diff
@@ -97,22 +105,89 @@ def run_docstyle_check(repository):
...
     core_api = CoreApi(cfg.stage.core_api_url, cfg.stage.core_api_token)
     jobs_api = JobsApi(cfg.stage.jobs_api_url, cfg.stage.jobs_api_token)
-    check_system(results, core_api, jobs_api)
+    results.stage = check_system(core_api, jobs_api)
+
+    core_api = CoreApi(cfg.prod.core_api_url, cfg.prod.core_api_token)
+    jobs_api = JobsApi(cfg.prod.jobs_api_url, cfg.prod.jobs_api_token)
+    results.production = check_system(core_api, jobs_api)
+
+    check_system(core_api, jobs_api)
```
Comment (jpopelka) [inline dashboard/src/dashboard.py]: Isn't this check redundant ?

File: dashboard/src/dashboard.py [hunk 4]
```diff
@@ -97,22 +105,89 @@ def run_docstyle_check(repository):
...
-        # run_pylint(repository)
-        # run_docstyle_check(repository)
-        pass
-    print(results)
+        clone_repository(repository)
+        run_pylint(repository)
+        run_docstyle_check(repository)
+        get_source_files(results, repository)
+        parse_pylint_results(results, repository)
+        parse_docstyle_results(results, repository)
```
Comment (jpopelka) [inline dashboard/src/dashboard.py]: It's OK, but similar as you changed the `check_system()` to not pass in and modify `results` I would do:

```python
  results.source_files[repository] = get_source_files(repository)
  results.repo_linter_checks[repository] = parse_pylint_results(repository)
  results.repo_docstyle_checks[repository] = parse_docstyle_results(repository)
```

but I'll leave this up to you.

Comment (tisnik) [reply dashboard/src/dashboard.py]: actually I like functional style proposed by you ;) TY

**Pull request merged**